File: C#/D&D Character/DndCharacter.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class Dice
{
    private static Random _rnd = new Random();

    public static int RollDice()
    {
        return _rnd.Next(1, 7);
    }

    public static List<int> DicePool(int numDice)
    {
        var dice = new List<int>();

        for (int i = 1; i <= numDice; i++)
        {
            dice.Add(RollDice());
        }

        return dice;
    }
}

public class DndCharacter
{
    public int Strength { get; }
    public int Dexterity { get; }
    public int Constitution { get; }
    public int Intelligence { get; }
    public int Wisdom { get; }
    public int Charisma { get; }
    public int Hitpoints { get; }

    public DndCharacter()
    {
        this.Strength = Ability();
        this.Dexterity = Ability();
        this.Constitution = Ability();
        this.Intelligence = Ability();
        this.Wisdom = Ability();
        this.Charisma = Ability();
        this.Hitpoints = 10 + Modifier(this.Constitution);
    }

    public static int Modifier(int score)
    {
        if (score < 3)
        {
            throw new ArgumentException("Ability scores must be at least 3");
        }

        else if (score > 18)
        {
            throw new ArgumentException("Ability scores can be at most 18");
        }

        return (int)Math.Floor((double)(score - 10) / 2);
    }

    public static int Ability()
    {
        var dice = Dice.DicePool(4).OrderByDescending(d => d).ToList();
        dice.RemoveAt(dice.Count - 1);
        return dice.Sum();
    }

    public static DndCharacter Generate()
    {
        return new DndCharacter();
    }

    public override string ToString()
    {
        string output = "";

        output += $"Strength:     {this.Strength,2}\n";
        output += $"Dexterity:    {this.Dexterity,2}\n";
        output += $"Constitution: {this.Constitution,2}\n";
        output += $"Intelligence: {this.Intelligence,2}\n";
        output += $"Wisdom:       {this.Wisdom,2}\n";
        output += $"Charisma:     {this.Charisma,2}\n";
        output += $"Hitpoints:    {this.Hitpoints,2}\n";

        return output;
    }
}

File: JavaScript/dnd-character.js
const rollDice = () => {
    return Math.floor(Math.random() * 6) + 1;
}

const dicePool = (numDice) => {
    let dice = [];

    for(let i = 1; i <= numDice; i++) {
        dice.push(rollDice());
    }

    return dice;
};

const sum = (array) => {
    return array.reduce((accumulator, currentValue) => {
        return accumulator + currentValue
    }, 0);
}

const abilityModifier = (n) => {
    if(n < 3) {
        throw new Error('Ability scores must be at least 3');
    }

    else if(n > 18) {
        throw new Error('Ability scores can be at most 18');
    }

    return Math.floor((n - 10) / 2);
};
  
class Character {
    static rollAbility() {
        let dice = dicePool(4).sort().reverse();
        dice.pop();
        return sum(dice);
    }

    constructor() {
        this._strength = Character.rollAbility();
        this._dexterity = Character.rollAbility();
        this._constitution = Character.rollAbility();
        this._intelligence = Character.rollAbility();
        this._wisdom = Character.rollAbility();
        this._charisma = Character.rollAbility();
        this._hitpoints = 10 + abilityModifier(this._constitution);
    }
  
    get strength() {
        return this._strength;
    }
  
    get dexterity() {
        return this._dexterity;
    }
  
    get constitution() {
        return this._constitution;
    }
  
    get intelligence() {
        return this._intelligence;
    }
  
    get wisdom() {
        return this._wisdom;
    }
  
    get charisma() {
        return this._charisma;
    }
  
    get hitpoints() {
        return this._hitpoints;
    }
}

const player = new Character();
console.log(player);